src/expr/proof_rule.h

```diff
@@ -1102,6 +1102,30 @@ enum class PfRule : uint32_t
   // ---------------------
   // Conclusion: (Q)
   INT_TRUST,
+  //======== Multiplication sign inference
+  // Children: none
+  // Arguments: (f1, ..., fk, m)
+  // ---------------------
+  // Conclusion: (=> (and f1 ... fk) (~ m 0))
+  // Where f1, ..., fk are variables compared to zero (less, greater or not equal), m is a monomial from these variables, and ~ is the comparison (less or greater) that results from the signs of the variables.
+  // All variables with even exponent in m should be given as not equal to zero while all variables with odd exponent in m should be given as less or greater than zero.
+  ARITH_MULT_SIGN,
+  //======== Multiplication with positive factor
```

Comment:
I'm actually curious if we can use Alex's ARITH_SCALE_SUM_UPPER_BOUNDS for these two rules.

Reply:
Note this would require a SCOPE to conclude the implication as needed.

Reply:
I think that the `ARITH_SCALE_SUM_UPPER_BOUNDS` rule does something different, though I can't completely follow the documentation. I'm pretty sure, though, that it only deals with linear terms and there is no multiplication between variables involved.
`ARITH_MULT_SIGN` on the other hand is explicitly about nonlinear monomials.
There may be corner cases where both are applicable, but not for the general case.

Reply:
Would be good to get thoughts from @alex-ozdemir.

Reply:
These are not captured by ARITH_SCALE_SUM_UPPER_BOUNDS. As Gereon suggests, ARITH_SCALE_SUM_UPPER_BOUNDS requires the scalars to be rational constants, and both ARITH_MULT_SIGN and ARITH_MULT_POS can apply to non-constant multiplicands.

If we want to get the arith rules down to a nicer core, there are two steps that make sense to me:

   1. make ARITH_MULT_SIGN a macro on top of ARITH_MULT_POS and ARITH_MULT_NEG.
   2. define an ARITH_SUM rule, let this:
```
// Children: ((><1 a1 b1), ..., (><n an bn))
// Arguments: none
// Conclusion: (>< (+ a1 ... an) (+ b1 ... bn))
// where each ><i is a comparator, and >< is the "sum" of those comparators.
```
and then we could redefine ARITH_SCALE_SUM_UPPER_BOUNDS as a macro on top of ARITH_SUM, ARITH_MULT_POS, and ARITH_MULT_NEG.

Probably no need to do either in this PR.